projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/templates/utility.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import os
+import re
+
+
+def get_staging_table_name(target_schema, target_table):
+    """
+    Extracts staging table by given target schema and table.
+    """
+    staging_table_name = f"vdk_check_{target_schema}_{target_table}"
+
+    if len(staging_table_name) > 128:
+        raise ValueError(
+            f"Staging table - {staging_table_name} exceeds the 128 character limit."
+        )
+    return staging_table_name
+
+
+def get_file_content(sql_files_folder, sql_file_name):
+    """
+    Reads and returns file content by given path and file name.
+    """
+    query_full_path = os.path.join(sql_files_folder, sql_file_name)
+    with open(query_full_path) as query:
+        content = query.read()
+        return content
+
+
+def align_stg_table_with_target(target_table, stg_table, job_input):
+    """
+    Aligns structure of a given staging table with the structure of a given target table.
+    """
+    _create_table_like(target_table, stg_table, job_input)
+
+    orig_create_table_statement = _extract_create_table_statement(
+        target_table, job_input
+    )
+    clone_create_table_statement = _extract_create_table_statement(stg_table, job_input)
+
+    if orig_create_table_statement != clone_create_table_statement:
+        job_input.execute_query(f"DROP TABLE {stg_table}")
+        _create_table_like(target_table, stg_table, job_input)
+
+
+def _extract_create_table_statement(table_name, job_input):
+    statement = job_input.execute_query(f"SHOW CREATE TABLE {table_name}")[0][0]
+    statement = _remove_location(statement)
+    statement = _remove_table_properties(statement)
+    statement = _remove_create_table_prefix(statement)
+    return statement
+
+
+def _create_table_like(orig_table_name, clone_table_name, job_input):
```

Comment:
I wonder if this works with partitions. Would the partition columns be created properly. I cannot recall. Might be worth adding a TODO if the answer is NO, or you cannot check it easily.

Reply:
I've checked in the [documentation](https://docs.cloudera.com/documentation/enterprise/6/6.3/topics/impala_create_table.html) and it says: "If the original table is partitioned, the new table inherits the same partition key columns. Because the new table is initially empty, it does not inherit the actual partitions that exist in the original one. To create partitions in the new table, insert data or issue ALTER TABLE ... ADD PARTITION statements"